src/api/Feed.js

```diff
@@ -405,27 +418,57 @@ class Feed extends Base {
         const tasksPromise = shouldShowTasks ? this.fetchTasksNew() : Promise.resolve();
         const appActivityPromise = shouldShowAppActivity ? this.fetchAppActivity(permissions) : Promise.resolve();
 
-        Promise.all([
-            versionsPromise,
-            currentVersionPromise,
-            commentsPromise,
-            tasksPromise,
-            appActivityPromise,
-            annotationsPromise,
-        ]).then(([versions: ?FileVersions, currentVersion: ?BoxItemVersion, ...feedItems]) => {
-            const versionsWithCurrent = currentVersion
-                ? this.versionsAPI.addCurrentVersion(currentVersion, versions, this.file)
-                : undefined;
-            const sortedFeedItems = sortFeedItems(versionsWithCurrent, ...feedItems);
-            if (!this.isDestroyed()) {
-                this.setCachedItems(id, sortedFeedItems);
-                if (this.errors.length) {
-                    errorCallback(sortedFeedItems, this.errors);
-                } else {
-                    successCallback(sortedFeedItems);
+        const fileActivitiesPromise = shouldUseUAA
+            ? this.fetchFileActivities(permissions, [
+                  UAA_ACTIVITY_TYPE_ANNOTATION,
+                  UAA_ACTIVITY_TYPE_APP_ACTIVITY,
+                  UAA_ACTIVITY_TYPE_COMMENT,
+                  UAA_ACTIVITY_TYPE_TASK,
+              ])
+            : Promise.resolve();
+
+        if (shouldUseUAA) {
+            Promise.all([versionsPromise, currentVersionPromise, fileActivitiesPromise]).then(
```

Comment:
Lots of nesting here. Can we break each branch of logic into separate methods?

Reply:
Thought it broken down enough after splitting the success callback and error callback. Eventually the FF conditional `shouldUseUAA` will go away, leaving it with the `all` array and the `then` statement. Are you suggesting to create methods for `all` and `then`?

Reply:
My thought was each branch of the if/else could be its own method, but it's probably not necessary given the short lifetime of the else branch. Feel free to disregard.